test/unit/org/apache/cassandra/cql3/validation/operations/AlterTest.java

```diff
@@ -497,20 +498,50 @@ public void testDefaultRF() throws Throwable
+        synchronized(CassandraRelevantProperties.DATACENTER_SKIP_NAME_VALIDATION)
+        {
+            // Create a keyspace with expected DC name.
+            createKeyspace("CREATE KEYSPACE %s WITH replication = {'class' : 'NetworkTopologyStrategy', '" + DATA_CENTER + "' : 2 }");
 
-        // try modifying the keyspace
-        assertAlterKeyspaceThrowsException(ConfigurationException.class,
-                                           "Unrecognized strategy option {INVALID_DC} passed to NetworkTopologyStrategy for keyspace " + currentKeyspace(),
-                                           "ALTER KEYSPACE %s WITH replication = { 'class' : 'NetworkTopologyStrategy', 'INVALID_DC' : 2 }");
+            // try modifying the keyspace
+            assertAlterKeyspaceThrowsException(ConfigurationException.class,
+                                               "Unrecognized strategy option {INVALID_DC} passed to NetworkTopologyStrategy for keyspace " + currentKeyspace(),
+                                               "ALTER KEYSPACE %s WITH replication = { 'class' : 'NetworkTopologyStrategy', 'INVALID_DC' : 2 }");
+
+            alterKeyspace("ALTER KEYSPACE %s WITH replication = {'class' : 'NetworkTopologyStrategy', '" + DATA_CENTER + "' : 3 }");
 
-        alterKeyspace("ALTER KEYSPACE %s WITH replication = {'class' : 'NetworkTopologyStrategy', '" + DATA_CENTER + "' : 3 }");
+            // Mix valid and invalid, should throw an exception
+            assertAlterKeyspaceThrowsException(ConfigurationException.class,
+                                               "Unrecognized strategy option {INVALID_DC} passed to NetworkTopologyStrategy for keyspace " + currentKeyspace(),
+                                               "ALTER KEYSPACE %s WITH replication={ 'class' : 'NetworkTopologyStrategy', '" + DATA_CENTER + "' : 2 , 'INVALID_DC': 1}");
+        }
+    }
 
-        // Mix valid and invalid, should throw an exception
-        assertAlterKeyspaceThrowsException(ConfigurationException.class,
-                                           "Unrecognized strategy option {INVALID_DC} passed to NetworkTopologyStrategy for keyspace " + currentKeyspace(),
-                                           "ALTER KEYSPACE %s WITH replication={ 'class' : 'NetworkTopologyStrategy', '" + DATA_CENTER + "' : 2 , 'INVALID_DC': 1}");
+    /**
+     * Test that when cassandra.dc_skip_name_validation=true nothing is thrown when altering a keyspace to invalid DC option in replication configuration.
+     */
+    @Test
+    public void testAlterKeyspaceWithNTSAcceptsAnyDataCenterNamesIfValidationIgnored() throws Throwable
+    {
+        synchronized(CassandraRelevantProperties.DATACENTER_SKIP_NAME_VALIDATION)
+        {
+            try
+            {
+                CassandraRelevantProperties.DATACENTER_SKIP_NAME_VALIDATION.setBoolean(true);
```

Comment:
5.0 has a test utility, called `WithProperty` that you might find useful here. It's not very widely used, but it is kind of nice.

Example use (there are others):
https://github.com/datastax/cassandra/blob/919c83bf42a9e54676d3d76129021898e73987d7/test/unit/org/apache/cassandra/security/CryptoProviderTest.java#L86

For instance, this might be something like:
```
try (WithProperties properties = new WithProperties().setBoolean(DATACENTER_SKIP_NAME_VALIDATION, true))
```